### PR TITLE
added view budget histories component

### PR DESCRIPTION
**Pull request by elraphty**
## Describe your changes

This PR adds the view of the budget payment history component

## Issue ticket number and link

[692](https://github.com/stakwork/sphinx-tribes/issues/692)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
@elraphty can you update this pr to say "Closes #692"? Let's use this format moving forward. Thanks!

File: db/structs.go [hunk 1]
```diff
@@ -460,14 +461,33 @@ type BudgetStoreData struct {
 	Created      *time.Time `json:"created"`
 }
 
+type BudgetPaymentType string
+
+const (
+	Add      BudgetPaymentType = "add"
```
Comment (kevkevinpal) [inline db/structs.go:467]: I think deposit might be better wording here

File: db/structs.go [hunk 2]
```diff
@@ -460,14 +461,33 @@ type BudgetStoreData struct {
...
+type BudgetHistoryData struct {
+	ID          uint              `json:"id"`
+	OrgUuid     string            `json:"org_uuid"`
+	Amount      uint              `json:"amount"`
+	PaymentType BudgetPaymentType `json:"payment_type"`
+	Status      bool              `json:"status"`
+	SenderName  string            `json:"sender_name"`
+	Created     *time.Time        `json:"created"`
+	Updated     *time.Time        `json:"updated"`
 }
```
Comment (kevkevinpal) [inline db/structs.go:491]: how is this any different than BudgetHistory?

File: frontend/app/src/people/widgetViews/OrganizationDetails.tsx [hunk 3]
```diff
@@ -260,14 +262,20 @@ const OrganizationDetails = (props: { close: () => void; org: Organization | und
...
   const generateInvoice = async () => {
     const token = ui.meInfo?.websocketToken;
     if (token) {
       const data = await main.getBudgetInvoice({
         amount: amount,
         sender_pubkey: ui.meInfo?.owner_pubkey ?? '',
         org_uuid: uuid,
-        websocket_token: token
+        websocket_token: token,
+        payment_type: 'add'
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/OrganizationDetails.tsx:278]: change this to deposit

**Comment by elraphty** [comment]
> @elraphty can you update this pr to say "Closes #692"? Let's use this format moving forward. Thanks!

Okay, thanks @ecurrencyhodler, I didn't add closes because, I'm still working on the link for the bounty on payment history, which requires a refactor to the database query and payment history API, once I'm done with it, I will add "Closes to the next PR".

File: db/structs.go (hunk 2 above)
Comment (elraphty) [reply db/structs.go:491]: > how is this any different than BudgetHistory?

@kevkevinpal the SenderName column, is the difference, It is not necessary on the budget History table, so I created a new struct for it.

File: frontend/app/src/people/widgetViews/OrganizationDetails.tsx (hunk 3 above)
Comment (elraphty) [reply frontend/app/src/people/widgetViews/OrganizationDetails.tsx:278]: Okay, thanks.

**Comment by ecurrencyhodler** [comment]
> > @elraphty can you update this pr to say "Closes #692"? Let's use this format moving forward. Thanks!
> 
> Okay, thanks @ecurrencyhodler, I didn't add closes because, I'm still working on the link for the bounty on payment history, which requires a refactor to the database query and payment history API, once I'm done with it, I will add "Closes to the next PR".

Okay great! SGTM! Also the more you can provide context like that in the description, the better it will help us keep track of how things are split up.

File: db/structs.go [hunk 4]
```diff
@@ -460,14 +461,27 @@ type BudgetStoreData struct {
...
+	SenderPubKey string            `json:"sender_pubkey"`
+	Created      *time.Time        `json:"created"`
+	Updated      *time.Time        `json:"updated"`
+	Status       bool              `json:"status"`
+	PaymentType  BudgetPaymentType `json:"payment_type"`
+}
+
+type BudgetHistoryData struct {
+	BudgetHistory
+	SenderName string `json:"sender_name"`
```
Comment (kevkevinpal) [inline db/structs.go:484]: why wouldnt we just fetch from the People table to get the sender name or just include this in BudgetHistory?

Comment (kevkevinpal) [reply db/structs.go:484]: oh wait I guess this isnt a new table so this should be ok

**Comment by kevkevinpal** [comment]
merging nice work!

**Pull request merged**